docs/src/rule_author/example.md

```diff
@@ -0,0 +1,88 @@
+# Pedagogical Example
+
+This pedagogical example will show you how to write an `rrule` for a moderately interesting primal calculation.
+See [On writing good `rrule` / `frule` methods](@ref) section for more tips and gotchas.
+If you want to learn about `frule`s, you should still read and understand this example as many concepts are shared, and then look for real world `frule` examples in ChainRules.jl.
+
+## The primal
+
+We define a struct `Foo`
+```julia
+struct Foo
+    A::Matrix
+    c::Float64
+end
+```
+and a function that multiplies `Foo` with an `AbstractArray`:
+```julia
+function foo_mul(foo::Foo, b::AbstractArray)
+    return foo.A * b
+end
+```
+Note that field `c` is ignored in the calculation.
+
+## The `rrule`
+
+A common gotcha is forgetting to `import` the `rrule`
+```julia
+import ChainRulesCore: rrule
+```
+Then, we are able to extend it with a method for our use case
+```julia
+function rrule(::typeof(foo_mul), foo::Foo, b::AbstractArray)
```

Comment:
should we be encouraging explicit extension with `ChainRulesCore.rrule` (rather than relying on the behaviour of `import`)?